api/helpers.go

```diff
@@ -4,42 +4,34 @@ import (
 	"encoding/hex"
 	"encoding/json"
 	"errors"
-	"fmt" // required for evm encoding
+	"fmt"
 	"math/big"
 	"reflect"
-	"strings"
 
 	cmtpool "github.com/cometbft/cometbft/mempool"
 	ctypes "github.com/cometbft/cometbft/rpc/core/types"
 	"github.com/ethereum/go-ethereum/accounts/abi"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/iancoleman/strcase"
 	"go.vocdoni.io/dvote/types"
+	"go.vocdoni.io/dvote/vochain/indexer/indexertypes"
 	"go.vocdoni.io/proto/build/go/models"
 	"google.golang.org/protobuf/encoding/protojson"
 	"google.golang.org/protobuf/proto"
 )
 
-func (a *API) electionSummaryList(pids ...[]byte) ([]*ElectionSummary, error) {
-	processes := []*ElectionSummary{}
-	for _, pid := range pids {
-		// TODO(mvdan): We construct ElectionSummaries in many places, deduplicate.
-		// TODO(mvdan): ProcessInfo could give us the results envelope height as well.
-		procInfo, err := a.indexer.ProcessInfo(pid)
-		if err != nil {
-			return nil, ErrCantFetchElection.WithErr(err)
-		}
-		processes = append(processes, &ElectionSummary{
-			ElectionID:     procInfo.ID,
-			OrganizationID: procInfo.EntityID,
-			Status:         strings.ToLower(models.ProcessStatus_name[procInfo.Status]),
-			StartDate:      procInfo.CreationTime,
-			EndDate:        a.vocinfo.HeightTime(int64(procInfo.EndBlock)),
-			FinalResults:   procInfo.FinalResults,
-			VoteCount:      procInfo.VoteCount,
-		})
+func (a *API) electionSummary(pi *indexertypes.Process) ElectionSummary {
+	return ElectionSummary{
+		ElectionID:     pi.ID,
+		OrganizationID: pi.EntityID,
+		// XXX: some callers used strings.ToLower on Status; what should we consolidate on?
```

Comment:
I think we should consolidate it to the name defined in the protobuf file. So it if the enum is using capital letters, we should return capital letters. The consumers of the API use the same protobuf file.

Reply:
Gui appears to have started a new thread about this, so to consolidate the two, I'll copy your response there and respond to both there.